### PR TITLE
pending style cops 182027741

### PR DESCRIPTION
**Pull request by JoeCohen**
This PR sets up Style department cops that are pending as of Rubocop 1.31.1:
- Configures the cops (explicitly enables/disables each of them in `.rubocop.yml`
- Fixes offenses or disables them inline in the code

This PR delivers part of https://www.pivotaltracker.com/story/show/182027741

### CODE REVIEW

**Comment by JoeCohen** [comment]
Tests pass locally

In the push check the test suite fails:
```
Progress: |==== FAIL CacheTest#test_propagate_lifeform (22.08s)
        Expected Mon, 18 Jul 2022 12:42:06.000000000 EDT -04:00 to be < Mon, 18 Jul 2022 12:41:27.540448966 EDT -04:00.
        test/models/cache_test.rb:[10](https://github.com/MushroomObserver/mushroom-observer/runs/7393947427?check_suite_focus=true#step:13:11)7:in `block in test_propagate_lifeform'
        test/models/cache_test.rb:105:in `test_propagate_lifeform'
```
In the pull_request check the test passes, but the check fails because Coveralls times out.
```
Run bundle exec rails test
Started with run options --seed 3[6](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:13:7)259
Progress: |=================================================================grep: /home/runner/work/mushroom-observer/mushroom-observer/log/test.log: No such file or directory
===|
Finished in 243.51180s
19[7](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:13:8)[8](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:13:9) tests, 27[9](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:13:10)33 assertions, 0 failures, 0 errors, 0 skips
Lcov style coverage report generated for Minitest to /home/runner/work/mushroom-observer/mushroom-observer/coverage/lcov/lcov.info
[12](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:13:13)s

Run coverallsapp/github-action@master
Using lcov file: ./coverage/lcov/lcov.info
Error: Bad response: 504 <html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h[1](https://github.com/MushroomObserver/mushroom-observer/runs/7394175037?check_suite_focus=true#step:14:1)>504 Gateway Time-out</h1></center>
</body>
</html>
```
I will try to have a look this afternoon (PDT). I'm tied up until then.

**Comment by JoeCohen** [comment]
Huh. I'm so mixed up. Now it 
- Fails locally. (Actually a big improvement, as maybe I can debug it locally.);
- Fails the push check
- Passes the pull_request check.

File: app/classes/map_set.rb [hunk 1]
```diff
@@ -119,10 +119,7 @@ def east_west_distance
   def update_extents_with_point(loc)
     lat = loc.lat.to_f.round(4)
     long = loc.long.to_f.round(4)
-    if !@north
-      @north = @south = lat
-      @east = @west = long
-    else
+    if @north
```
Comment (JoeCohen) [inline app/classes/map_set.rb:122]: Here (and in many other places), the `if` and `else` code is swapped in order to use a positive condition with `if`.
All of these have been autocorrected, so there won't be any transcription or other human errors. 
Rubocop says the cop and autocorrection are safe, which makes sense to me.
See Style/NegatedIfElseCondition, https://docs.rubocop.org/rubocop/1.31/cops_style.html#stylenegatedifelsecondition

File: config/consts.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ class ImageConfigData
 
   def initialize
     root = File.expand_path("..", __dir__)
-    @env = ENV["RAILS_ENV"] || "development"
+    @env = ENV.fetch("RAILS_ENV", "development")
```
Comment (JoeCohen) [inline config/consts.rb:10]: I think this is the cop in this PR which results in longer code. It's safer this way.
See Style/FetchEnvVar https://docs.rubocop.org/rubocop/1.31/cops_style.html#stylefetchenvvar

File: script/refresh_name_lister_cache [hunk 3]
```diff
@@ -100,6 +100,8 @@ end.flatten
 unless File.directory?(MO.name_lister_cache_file)
   FileUtils.mkpath(File.dirname(MO.name_lister_cache_file))
 end
+# NOTE: Use File.write in the following, or at least disable Style/FileWrite
+# JDC 2022-07-16
 File.open(MO.name_lister_cache_file, "w") do |fh|
```
Comment (JoeCohen) [inline script/refresh_name_lister_cache:105]: I can't immediately get my head around how this is parsed.

File: test/fixtures/names.yml [hunk 4]
```diff
@@ -1249,6 +1249,8 @@ eukarya:
 # This should actually be section Leucoagaricus, but I was too lazy to add the genus Leucoagaricus!
 sect_agaricus:
   <<: *DEFAULTS
+  created_at: 2007-06-23 20:04:00
+  updated_at: 2007-06-23 20:04:00
```
Comment (JoeCohen) [inline test/fixtures/names.yml:1253]: I cherry-picked this change in order to fix intermittent test failures on this branch.

**Comment by JoeCohen** [comment]
1. Another mind-numbing PR in order to avoid falling further behind in rubocopping.
2. If you hate any of the new cops, I can globally disable them, globally adjust configurable attributes, and/or disable them in some places in the code.

File: test/models/image_script_test.rb [hunk 5]
```diff
@@ -174,39 +174,17 @@ def mysql_config_file
...
+    File.write("#{local_root}/orig/#{in_situ_id}.jpg", "B")
+    File.write("#{local_root}/1280/#{in_situ_id}.jpg", "C")
+    File.write("#{local_root}/960/#{in_situ_id}.jpg", "D")
+    File.write("#{local_root}/640/#{in_situ_id}.jpg", "E")
+    File.write("#{local_root}/320/#{in_situ_id}.jpg", "F")
+    File.write("#{local_root}/thumb/#{in_situ_id}.jpg", "G")
+    File.write("#{local_root}/960/#{turned_over_id}.jpg", "H")
+    File.write("#{local_root}/640/#{turned_over_id}.jpg", "I")
+    File.write("#{local_root}/320/#{turned_over_id}.jpg", "J")
+    File.write("#{local_root}/thumb/#{turned_over_id}.jpg", "K")
```
Comment (pellaea) [inline test/models/image_script_test.rb:187]: Nice!  I wish I'd known about File.write() from the start.  How much cleaner and more compact!

**Review by pellaea — APPROVED** [review]
I gave up and started just spot checking after a point.  This all looks totally fine to me.

**Review by nimmolo — APPROVED** [review]
Cursory review, but i trust you!

**Comment by JoeCohen** [comment]
Thank you both for looking at this. It would put anyone to sleep.
I will merge.

**Pull request merged**